src/main/java/net/minecraftforge/event/RegisterGameTestsEvent.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.event;
+
+import net.minecraft.gametest.framework.GameTest;
+import net.minecraft.gametest.framework.GameTestGenerator;
+import net.minecraft.gametest.framework.GameTestServer;
+import net.minecraftforge.eventbus.api.Event;
+import net.minecraftforge.fml.event.IModBusEvent;
+import net.minecraftforge.gametest.ForgeGameTestHooks;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Game tests are registered on client or server startup.
+ * It is only run once for a given instance of the game if {@link ForgeGameTestHooks#isGametestEnabled} returns true.
+ * This is the preferred way to register your game tests.
+ * <p>
+ * Fired on the Mod bus, see {@link IModBusEvent}.
+ */
+public class RegisterGameTestsEvent extends Event implements IModBusEvent
+{
+    private final Set<String> enabledNamespaces;
+    private final Set<Method> gameTestMethods;
+    private final Set<Method> gameTestMethodsView;
+
+    public RegisterGameTestsEvent(Set<String> enabledNamespaces)
+    {
+        this.enabledNamespaces = new HashSet<>(enabledNamespaces);
+        this.gameTestMethods = new HashSet<>();
+        this.gameTestMethodsView = Collections.unmodifiableSet(this.gameTestMethods);
+    }
+
+    /**
+     * Returns the mutable set of enabled namespaces.
+     * If a game test's template does not have a namespace inside this set, it will not be registered.
+     * An empty set means all namespaces are enabled.
+     * By default, this can only be non-empty when running through {@link GameTestServer}.
+     *
+     * @return the mutable set of enabled namespaces
+     */
+    public Set<String> getEnabledNamespaces()
+    {
+        return enabledNamespaces;
+    }
+
+    /**
+     * Returns an immutable view of all registered gametest methods.
+     *
+     * @return an immutable view of all registered gametest methods
+     */
+    public Set<Method> getGameTestMethods() {
+        return this.gameTestMethodsView;
+    }
```

Comment:
I'm not sure if we should be exposing the registered methods as public API, as I'd personally prefer this to be made an implementation detail. 
You could refactor this event such that the caller of the constructor passes in a mutable set controlled by that caller, which this event add the methods to.
(If you decide against this suggestion, you'll need to resolve the brace being on the same line. :shipit:)